### ISSUE TITLE
full compatibility with node events api

### ISSUE THREAD

**Issue by jasonkarns** [r-park/event-signal#1]
Would you be interested in a PR that adds the full complement of node's EventEmitter API?

Specifically, aliases for `on` and `once`? (And ideally, `defaultMaxListeners` and `setMaxListeners` as well)

**Comment by r-park** [comment]
PR for `defaultMaxListeners` and `setMaxListeners` would be welcome.

I think `on` would be syntactically awkward in this case, given the intended use pattern:
```javascript
var obj = {};
obj.updated = new EventSignal()
obj.updated.on(function(){})
```
I've been thinking about adding aliases `add` and `remove` which I think would be more readable, albeit not in line with node's api:
```javascript
obj.updated
  .add(function listener(){})
  .emit()
  .remove(listener)
```

**Comment by jasonkarns** [comment]
After using your lib a little, I realized I needed more control over how
the listeners were handled. So I created my own flavor wherein each signal
is itself an event emitter (and can be optionally provided to the
constructor). Thus, each signal implements the full API (dy definition) and
also allows each signal to share a common eventEmitter instance. (Which was
useful in my particular case.)

All that to say, thanks for having this library. It gave me the idea for
our implementation. But I won't be able to submit in work or PRs here.

Thanks!
